### ISSUE TITLE
False positive "Use of moved value" 

### ISSUE THREAD

**Issue by arctic-alpaca** [intellij-rust/intellij-rust#10538]
## Environment

* **IntelliJ Rust plugin version:** 0.4.196.5399-231-nightly
* **Rust toolchain version:** rustc 1.70.0 (90c541806 2023-05-31)
* **IDE name and version:** CLion 2023.1.3
* **Operating system:** Fedora 37

## Problem description
![Intellij_repro_cut](https://github.com/intellij-rust/intellij-rust/assets/67190338/70e12924-1efd-4fb7-b765-b463bb34f97e)

The reported "Use of moved value" error is incorrect. The issue seems to be caused by `(test.first, test.second) = do_stuff();`.

There is also https://github.com/intellij-rust/intellij-rust/issues/6826 but I'm not sure if it is caused by the same underlying problem.
## Steps to reproduce

```rust
pub struct Test {
    pub(crate) first: usize,
    pub(crate) second: Vec<usize>,
}

pub fn do_stuff() -> (usize, Vec<usize>) {
    (1, vec![1, 2, 3, 4])
}
pub fn do_more_stuff() -> Test {
    let mut test = Test {
        first: 1,
        second: vec![],
    };
    (test.first, test.second) = do_stuff();
    test
}

fn main() {
    println!("Hello, world!");
}
```

**Comment by gimbles** [comment]
+1, also facing. :(
![image](https://github.com/intellij-rust/intellij-rust/assets/93856041/ec86d402-74be-46e2-ab05-bb155e922a3d)

**Comment by arctic-alpaca** [comment]
@gimbles Your issue might be different as it seems related to `Copy` types. If so, it should be fixed in the nightly version of the Rust plugin: https://github.com/intellij-rust/intellij-rust/issues/10427

**Comment by vlad20012** [comment]
@lilith13666, your issue is different. Please, take a look at #10427 or file new issue

**Comment by lilith13666** [comment]
> @lilith13666, your issue is different. Please, take a look at #10427 or file new issue

I don't see in what way my issue is different. I'm getting this "Use of moved value" in so many different places.
Anyway, the issue you mentioned is closed, so what's the point?
Is there any way to find out the underlying rule is being applied to each individual false positive like this?

**Comment by Maksim20023** [comment]
Hi @arctic-alpaca! Thank you for the report.
The issue is related to the false-positive "use of moved value" warning, and it is already known to us: #10505  You can subscribe to this issue to stay informed and vote (click the emoji button in the bottom-left corner of the GitHub post).
If you have any further suggestions, please feel free to reach out.

@lilith13666 
Could you please specify if the issue still occurs on the latest Rust plugin version?

**Comment by lilith13666** [comment]
@Maksim20023 The issue is resolved. Thanks!

**Comment by arctic-alpaca** [comment]
@Maksim20023 Sorry I missed the existing issue, thanks for linking it!
I'll close this as it is a duplicate.
